lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -281,17 +281,22 @@ public function cond($if, $then, $else)
     }
 
     /**
-     * Ensures an array or operator expression is converted to an array.
+     * Converts an expression object into an array, recursing into nested items
+     *
+     * For expression objects, it calls getExpression on the expression object.
+     * For arrays, it recursively calls itself for each array item. Other values
+     * are returned directly
      *
      * @param mixed|self $expression
-     * @return mixed
+     * @return string|array
+     * @internal
      */
-    protected function ensureArray($expression)
+    public static function convertExpression($expression)
     {
         if (is_array($expression)) {
             $array = [];
             foreach ($expression as $index => $value) {
-                $array[$index] = $this->ensureArray($value);
+                $array[$index] = static::convertExpression($value);
```

Comment:
Is `$expression` always a compact array in this case, or might it be associative? I'm just curious if `$index` is relevant here or if `$array[]` would be functionally equivalent.